src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -608,30 +607,29 @@ public override bool Equals(object comparand)
 
         public override int GetHashCode()
         {
-            // For IPv6 addresses, we cannot simply return the integer
-            // representation as the hashcode. Instead, we calculate
-            // the hashcode from the address literals and ScopeId.
-            if (IsIPv6)
+            if (_hashCode != 0)
```

Comment:
0 is a valid value. I think that's likely ok since it would just trigger recomputation.